packages/starknet-snap/src/utils/starknetUtils.ts

```diff
@@ -214,13 +235,19 @@ export const estimateFeeBulk = async (
   invocationsDetails?: UniversalDetails,
   cairoVersion?: CairoVersion,
 ): Promise<EstimateFee[]> => {
-  return getAccountInstance(
+  // Use nonce from invocationsDetails or get it from the account nonce on the latest accepted block to avoid issues with pending transactions.
+  // as estimating fees on the pending block can fail if there are already transactions in the pending state.
+  const nonce =
+    invocationsDetails?.nonce ??
+    (await getAccountNonceOnLatest(network, senderAddress, privateKey));
+  return await getAccountInstance(
```

Comment:
actually i just find out there is a easier way to do that

by update the getProvider to 
```
export const getProvider = (network: Network, blockIdentifier): ProviderInterface => {
  let providerParam: ProviderOptions = {};
  providerParam = {
    nodeUrl: getRPCUrl(network.chainId),
  };
  if (blockIdentifier) {
providerParam.blockIdentifier = blockIdentifier
}
  return new Provider(providerParam);
};
```
in getAccountInstance
```
export const getAccountInstance = (
  network: Network,
  userAddress: string,
  privateKey: string | Uint8Array,
  cairoVersion?: CairoVersion,
  blockIdentifier?: blockIdentifier
): Account => {
  const provider = getProvider(network, blockIdentifier);
  return new Account(
    provider,
    userAddress,
    privateKey,
    cairoVersion ?? CAIRO_VERSION,
  );
};
```
WDYT

Reply:
That would be great. Can we confirm that the `new Account` uses the blockIdentifier of the provider ?

Reply:
Double checked works like a charm.

Reply:
yea, it is actually depends on the provider

so we setup the config of a provider
then we pass provider as an arg to account
when account doing the api, it will use the provider options if no nonce pass in